db/migrate/20181016140921_migrate_orch_stacks_to_have_ownership_concept.rb

```diff
@@ -0,0 +1,83 @@
+class MigrateOrchStacksToHaveOwnershipConcept < ActiveRecord::Migration[5.0]
+  class ExtManagementSystem < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+
+    def tenant_identity
+      User.super_admin
+    end
+  end
+
+  class Service < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+    belongs_to :evm_owner, :class_name => "User"
+
+    def tenant_identity
+      user = evm_owner
+      user = User.super_admin if user.nil? || !user.miq_group_ids.include?(miq_group_id)
+      user
+    end
+  end
+
+  class MiqGroup < ApplicationRecord
+    belongs_to :tenant
+  end
+
+  class User < ApplicationRecord
+    has_and_belongs_to_many :miq_groups
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+
+    def self.super_admin
+      User.find_by(:userid => "admin")
+    end
+
+    def current_tenant
+      current_group.tenant
+    end
+
+    def current_group
+      miq_groups.first
+    end
+  end
+
+  class ServiceResource < ApplicationRecord
+    belongs_to :service
+  end
+
+  class Tenant < ApplicationRecord
+  end
+
+  class OrchestrationStack < ApplicationRecord
+    self.inheritance_column = :_type_disabled # disable STI
+    has_many :service_resources, :as => :resource
+    has_many :direct_services, :through => :service_resources, :source => :service
+    belongs_to :ext_management_system, :foreign_key => :ems_id
+
+    def service
+      direct_services.first
+    end
+  end
+
+  def up
+    say_with_time("Migrating existing orchestration stacks to have direct owners, groups, tenant") do
+      OrchestrationStack.select { |orch_stack| orch_stack.service.present? }.each do |stack|
```

Comment:
Since you are trying to set all OrchestrationStack instance it might be better to loop over all and handle the three cases.  With service, no service but ems, or neither.